src/JustSaying/Fluent/ISubscriptionBuilder`1.cs

```diff
@@ -26,7 +26,9 @@ internal interface ISubscriptionBuilder<out T>
         void Configure(
             JustSayingBus bus,
             IHandlerResolver resolver,
+            IServiceResolver serviceResolver,
```

Comment:
Yeah I agree that having both resolvers plus the ServicesBuilder here isn't ideal. We could perhaps introduce an `IMiddlewareResolver`, though that adds a bit more machinery for questionable gain.

The ServicesBuilder is needed because we special case the registration/resolution of `IMessageLockAsync` because of JustSaying's explicit support for ExactlyOnce. We could remove this dependency by removing the `WithMessageLock` API and just requiring the user to register an `IMessageLockAsync` into the DI container, which would be resolvable in the middleware.

I suspect we might need to keep `IHandlerResolver` and `IServiceResolver` though, as a custom `IHandlerResolver` could be provided even if the user uses a DI container - we can't just resolve the `IHandlerResolver` from the `IServiceResolver`

Reply:
I had a punt at doing this over here https://github.com/gkinsman/JustSaying/pull/1

Reply:
Considering this resolved 👍